scripts/list_boards.py

```diff
@@ -83,30 +146,192 @@ def find_arch2board_set_in(root, arches):
                 file_name = maybe_defconfig.name
                 if file_name.endswith('_defconfig'):
                     board_name = file_name[:-len('_defconfig')]
-                    ret[arch].add(Board(board_name, arch, maybe_board))
+                    ret[arch].add(Board(board_name, maybe_board, 'v1', arch=arch))
 
     return ret
 
+
+def find_v2_boards(args):
+    root_args = argparse.Namespace(**{'soc_roots': args.soc_roots})
+    systems = list_hardware.find_v2_systems(root_args)
+
+    boards = []
+    board_files = []
+    for root in args.board_roots:
+        board_files.extend(Path(root).rglob(BOARD_YML))
+
+    for board_yml in board_files:
+        if board_yml.is_file():
+            with board_yml.open('r') as f:
+                b = yaml.safe_load(f.read())
+
+            try:
+                pykwalify.core.Core(source_data=b, schema_data=board_schema).validate()
+            except pykwalify.errors.SchemaError as e:
+                sys.exit('ERROR: Malformed "build" section in file: {}\n{}'
+                         .format(board_yml.as_posix(), e))
+
+            board = b['board']
+            if args.board is not None:
+                if board['name'] not in args.board:
```

Comment:
```suggestion
                if board['name'] != args.board:
```
Unless you want to support building for multiple boards at once :) Try `-DBOARD=mps3_rzt2m_starter_kit`